src/Types.jl

```diff
@@ -236,6 +237,9 @@ end
 Base.:(==)(t1::Project, t2::Project) = all(x -> (getfield(t1, x) == getfield(t2, x))::Bool, fieldnames(Project))
 Base.hash(t::Project, h::UInt) = foldr(hash, [getfield(t, x) for x in fieldnames(Project)], init=h)
 
+# only hash the deps and compat fields as they are the only fields that affect a resolve
+project_resolve_hash(t::Project, h::UInt = UInt(0)) = string(foldr(hash, [t.deps, t.compat], init=h))
```

Comment:
Wouldn't this be better as a SHA1?

Reply:
ah ok. Would you mind helping me with a code suggestion?

Reply:
Maybe something like this?
```
julia> using SHA

julia> io = IOBuffer();

julia> write(io, "write everything in a canonical order");

julia> bytes2hex(sha1(seekstart(io)))
"a28efaf9173c5c61bfe48ed1402e2c3408a69b3b"
```

Reply:
thanks!